src/translator/self_adaptive.h

```diff
@@ -0,0 +1,189 @@
+#pragma once
+
+#include "common/config.h"
+#include "common/file_stream.h"
+#include "data/batch_generator.h"
+#include "data/text_input.h"
+#include "models/model_task.h"
+#include "training/scheduler.h"
+#include "training/validator.h"
+#include "translator/swappable.h"
+#include "data/adaptive_context.h"
+
+namespace marian {
+
+using namespace data;
+
+/**
+ * @breif Implementation of the self-adaptive translation mode.
+ *
+ * Self-adaptive translation means optionally using a set of context sentences
+ * (e.g., provided by a translation memory), that are similar to the
+ * translatable sentence, to train the model for a few iterations to fine-tune
+ * it before translating the given sentence.
+ */
+class TrainSelfAdaptive : public ModelTask, public ModelServiceTask {
+public:
+  TrainSelfAdaptive(Ptr<Options> options) : options_(options) {
+    options_->set("shuffle", "none");
+    // Disable early stopping because typically training would happen for only a few iterations and
+    // and also it doesn't make much sense to run the validation metrics on the validation dataset here
```

Comment:
```suggestion
    // Disable early stopping because typically training would happen for only a few iterations and
    // also it doesn't make much sense to run the validation metrics on the validation dataset here
```

Reply:
Is it possible to specify validation sets for marian-adaptive or is it disabled?